pkg/gatewayconfigurationserver/gatewayconfigurationserver.go

```diff
@@ -47,6 +49,56 @@ type GatewayConfigurationServer struct {
 	config *Config
 }
 
+func (gcs *GatewayConfigurationServer) makeGatewayConfigHandler(f func(context.Context, echo.Context, *ttnpb.Gateway) error) func(echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := gcs.getContext(c)
+		gtwID := c.Get(gatewayIDKey).(ttnpb.GatewayIdentifiers)
+		cc, err := gcs.GetPeerConn(ctx, ttnpb.ClusterRole_ENTITY_REGISTRY, nil)
+		if err != nil {
+			return err
+		}
+		client := ttnpb.NewGatewayRegistryClient(cc)
+		gtw, err := client.Get(ctx, &ttnpb.GetGatewayRequest{
+			GatewayIdentifiers: gtwID,
+			FieldMask: types.FieldMask{
+				Paths: []string{
+					"antennas",
+					"frequency_plan_id",
+					"gateway_server_address",
+				},
+			},
+		}, gcs.WithClusterAuth())
+		if err != nil {
+			return err
+		}
+		return f(ctx, c, gtw)
+	}
+}
+
+func (gcs *GatewayConfigurationServer) makeJSONGatewayConfigHandler(f func(context.Context, *ttnpb.Gateway) (interface{}, error)) func(echo.Context) error {
+	return gcs.makeGatewayConfigHandler(func(ctx context.Context, c echo.Context, gtw *ttnpb.Gateway) error {
+		msg, err := f(ctx, gtw)
+		if err != nil {
+			return err
+		}
+		return c.JSONPretty(http.StatusOK, msg, "\t")
+	})
+}
+
+func (gcs *GatewayConfigurationServer) makeTextMarshalerGatewayConfigHandler(contentType string, f func(context.Context, *ttnpb.Gateway) (encoding.TextMarshaler, error)) func(echo.Context) error {
```

Comment:
Nit: "TextMarshalling" is a golang term right? I think it would be nice to use the output file format in the funtion name. For example, `makeJSON`, `makeTOML`, `makeYAML` etc..

Reply:
It's named this way, because this method uses the `MarshalText` *method* on an `encoding.TextMarshaler`.
E.g. `makeJSONHandler` takes an `interface{}`

Reply:
I still think that it's better to name a function after what it creates rather than what method it calls internally. But again this is nitpick.

Reply:
This method does not know what it creates - that's the point.
The content-type is a parameter - it may create TOML, JSON, YAML or anything else.